debezium-core/src/main/java/io/debezium/cloudevents/CloudEventsConverter.java

```diff
@@ -0,0 +1,483 @@
+            case JSON:
+                enableJsonSchemas = ceConfig.cloudeventsJsonSchemasEnable();
+                break;
+            case AVRO:
+                if (this.dataSchemaRegistry != null) {
+                    avroDataConverter = new AvroConverter(dataSchemaRegistry);
+                }
+                dataSchemaRegistryUrls = ceConfig.cloudeventsDataSchemaRegistryUrls();
+                if (dataSchemaRegistryUrls == null) {
+                    throw new DataException("Need url(s) for schema registry instances for the data field of CloudEvents");
+                }
+                Map<String, Object> avroConfigs = new HashMap<>(configs);
+                avroConfigs.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, String.join(",", dataSchemaRegistryUrls));
+                avroDataConverter.configure(avroConfigs, false);
+                break;
+        }
+    }
+
+    @Override
+    public byte[] fromConnectData(String topic, Schema schema, Object value) {
+        if (schema == null || value == null) {
+            return null;
+        }
+        if (schema.type() != STRUCT) {
+            throw new DataException("Mismatching schema");
+        }
+
+        RecordParser parser = RecordParser.create(schema, value);
+        CloudEventsMaker maker = CloudEventsMaker.create(parser, ceSerializerType,
+                (dataSchemaRegistryUrls == null) ? null : String.join(",", dataSchemaRegistryUrls));
+
+        byte[] serializedData;
+        byte[] serializedCloudEvents;
+
+        switch (dataSerializerType) {
+            case JSON:
+                JsonNode node = convertToJsonNode(maker.ceDataAttributeSchema(), maker.ceDataAttribute(), enableJsonSchemas);
+                serializedData = jsonSerializer.serialize(topic, node);
+                break;
+            case AVRO:
+                serializedData = avroDataConverter.fromConnectData(topic, maker.ceDataAttributeSchema(), maker.ceDataAttribute());
+                break;
+            default:
+                throw new DataException("No such serializer for \"" + dataSerializerType + "\" format");
+        }
+
+        SchemaAndValue ceSchemaAndValue = convertToCloudEventsFormat(maker, serializedData);
+        switch (ceSerializerType) {
+            case JSON:
+                JsonNode node = convertToJsonNode(ceSchemaAndValue.schema(), ceSchemaAndValue.value(), false);
+                serializedCloudEvents = jsonSerializer.serialize(topic, node);
+                break;
+            case AVRO:
+                serializedCloudEvents = avroCEConverter.fromConnectData(topic, ceSchemaAndValue.schema(), ceSchemaAndValue.value());
+                break;
+            default:
+                throw new DataException("No such serializer for \"" + dataSerializerType + "\" format");
+        }
+
+        return serializedCloudEvents;
+    }
+
+    @Override
+    public SchemaAndValue toConnectData(String topic, byte[] value) {
```

Comment:
@Wang-Yu-Chao, I'm wondering: when would this method ever be invoked? The way I see it, this converter would only ever be used for serializing the Debezium events when writing *into* a Kafka topic. For the consuming side, I'd expect other clients then Kafka Connect which will be able to interpret the CloudEvents structure. When going from Kafka Connect source (Debezium) to another Kafka Connect sink (e.g. JDBC or Elasticsearch), using CloudEvents probably doesn't make that much sense. Or do you foresee a case where this would be useful?